modules/statistics/server/controllers/statistics.server.controller.js

```diff
@@ -12,9 +12,105 @@ var path = require('path'),
     User = mongoose.model('User');
 
 /**
+ * Get count of all public users
+ */
+exports.getUsersCount = function(callback) {
+  User.count({ public: true }, function(err, count) {
+    callback(err, count || 0);
+  });
+};
+
+/**
+ * Get count of all public users
+ */
+exports.getExternalSiteCount = function(site, callback) {
+  var validSites = ['bewelcome', 'couchsurfing', 'warmshowers', 'facebook', 'twitter', 'github'];
+
+  // Validate site
+  if (!site || validSites.indexOf(site) === -1) {
+    return callback(new Error('Missing external site id.'));
+  }
+
+  // Build the query
+  var query = { public: true };
+
+  switch (site) {
+    case 'bewelcome':
+      query.extSitesBW = { $exists: true, $ne: '' };
+      break;
+    case 'couchsurfing':
+      query.extSitesCS = { $exists: true, $ne: '' };
+      break;
+    case 'warmshowers':
+      query.extSitesWS = { $exists: true, $ne: '' };
+      break;
+    case 'facebook':
+      query.additionalProvidersData = {
+        facebook: { $exists: true }
+      };
+      break;
+    case 'twitter':
+      query.additionalProvidersData = {
+        twitter: { $exists: true }
+      };
+      break;
+    case 'github':
+      query.additionalProvidersData = {
+        github: { $exists: true }
+      };
+      break;
+  }
+
+  User.count(query, function(err, count) {
+    callback(err, count || 0);
+  });
+};
+
+/**
+ * Get count of hosting offers
+ * Callback will be called with Object `{ yes: Int, maybe: Int }`
+ */
+exports.getOffersCount = function(callback) {
+  Offer.aggregate({
+    $group: {
+      _id: '$status',
+      count: { $sum: 1 }
+    }
+  },
+  function(err, counters) {
+    var values = {
+      yes: 0,
+      maybe: 0
+    };
+    if (counters && counters.length > 0) {
+      counters.forEach(function(counter) {
+        if (['yes', 'maybe'].indexOf(counter._id) !== -1) {
+          values[counter._id] = counter.count || 0;
+        }
+      });
+    }
```

Comment:
@chmac yup, fair enough — I do think that any time anyone says "this might be unclear", it asks for a comment or simplifying the code. This kind of codebase should be crystal clear.